javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/matcher/ReferenceMatcher.java

```diff
@@ -160,9 +159,8 @@ private boolean doesMatch(ClassLoader loader) {
     Set<HelperReferenceWrapper.Method> plainMethods = new HashSet<>();
     collectMethodsFromTypeHierarchy(helperWrapper, abstractMethods, plainMethods);
 
-    Set<HelperReferenceWrapper.Method> unimplementedMethods =
-        Sets.difference(abstractMethods, plainMethods);
-    for (HelperReferenceWrapper.Method unimplementedMethod : unimplementedMethods) {
+    abstractMethods.removeAll(plainMethods);
```

Comment:
I think there is a change of behaviour here: in the new version you mutate `abstractMethods`, the old version did not that, did it?

Reply:
Yeah but it's a local variable that wasn't used after the call to difference. So I'm ok with it I could make a copy instead though